### PR TITLE
Split up enclave_server.h to a header and a source file

### PR DESCRIPTION
**Pull request by aalmos**
I have plans to do refactors with this class, as a first step I thought it'd be anyways worth splitting it to header and source.

### CODE REVIEW

File: oak/server/enclave_server.h [hunk 1]
```diff
@@ -168,11 +86,11 @@ class EnclaveServer final : public asylo::TrustedApplication {
...
   int port_;
 
-  std::unique_ptr<::oak::OakNode> node_;
-  std::shared_ptr<::grpc::ServerCredentials> credentials_;
+  std::unique_ptr<OakNode> node_;
+  std::shared_ptr<grpc::ServerCredentials> credentials_;
 
   ::grpc::AsyncGenericService module_service_;
-  std::unique_ptr<::grpc::ServerCompletionQueue> completion_queue_;
+  std::unique_ptr<grpc::ServerCompletionQueue> completion_queue_;
```
Comment (michael-kernel-sanders) [inline oak/server/enclave_server.h]: Please leave the :: prefix for namespaces other than std::.  You can add it in other places for consistency.

File: oak/server/enclave_server.h [hunk 2]
```diff
@@ -168,11 +86,11 @@ class EnclaveServer final : public asylo::TrustedApplication {
   // The port on which the server is listening.
   int port_;
 
-  std::unique_ptr<::oak::OakNode> node_;
-  std::shared_ptr<::grpc::ServerCredentials> credentials_;
+  std::unique_ptr<OakNode> node_;
```
Comment (michael-kernel-sanders) [inline oak/server/enclave_server.h]: @tiziano88 wanted to keep using ::oak:: for consistency, I think.  In general if we want to shorten these, I would prefer adding using directives.

**Comment by michael-kernel-sanders** [comment]
I was also considering refactoring this with OakNode... what did you have in mind?

File: oak/server/enclave_server.h (hunk 2 above)
Comment (tiziano88) [reply oak/server/enclave_server.h]: I don't particularly mind either way, but I would leave that change to a separate PR, perhaps at the same time as adding a "style guide" section to the README or CONTRIBUTING that explains what convention we are using going forward.

Comment (aalmos) [reply oak/server/enclave_server.h]: I was trying to follow here totw/151 with qualifying namespaces. I'd suggest following established Google best practices unless there's a good reason to not to do so.

Comment (michael-kernel-sanders) [reply oak/server/enclave_server.h]: Alright, totw/151 works for me.

**Comment by aalmos** [comment]
> I was also considering refactoring this with OakNode... what did you have in mind?

First I wanted to revise synchronization, because it's a messy. As per the asylo::TrustedApplication documentation, the only method that has to be thread safe is Run. Run calls GetServerAddress that has "EXCLUSIVE_LOCKS_REQUIRED(server_mutex_)", but it doesn't hold the mutex at all. And all the stuff like that. I'd also love to see the responsibilities of CreateServer outsourced to a static, or a completely internal method, and I'd like to dig around ModuleInvocation, as TBH that object smells a bit for me (an object that "recursively" allocates and deallocates itself).

**Comment by aalmos** [comment]
@tiziano88 @michael-kernel-sanders if you have no issues with this PR anymore, can you please merge it? Changes are actively coming into enclave_server.h and I keep fighting with conflicts :)

**Pull request merged**